commands/config/client_config.go

```diff
@@ -42,3 +45,34 @@ func NewClientConfig(c []byte) (*ClientConfig, error) {
 func (c *ClientConfig) GetProvidersMap() (map[string]ProviderConfig, error) {
 	return CreateProvidersMap(c.Providers)
 }
+
+func (c *ClientConfig) CheckKeyDir() (err error) {
+
+	var (
+		keyDir   string
+		fileInfo os.FileInfo
+	)
+
+	// if default key dir is not set we use the default logic
+	if c.KeyManagement.DefaultKeyDir == "" {
+		return
```

Comment:
I recognize this syntax of declaring variables in the output block and then letting return has some big advantages, but I don't want to use it in opkssh because it makes it harder to understand what is happening.